Add dev/deploy-site.sh script

cwsteinbach opened the pull request:
This patch adds dev/deploy-site.sh and updates the documentation in
site/README.md to reference this utility script.

rdblue commented on dev/deploy-site.sh:
```diff
@@ -0,0 +1,62 @@
+# If the following commands stop working, please check this ASF INFRA
+# page for updates: https://infra.apache.org/asfyaml-mkdocs.html
+cd ${projectdir}/site
+rm -r site
```
This will fail if `site` doesn't exist, right? Can you add a test to only remove the directory if it exists?

cwsteinbach replied:
Changed to `rm -rf`

jackye1995 commented on site/README.md:
````diff
@@ -52,7 +52,7 @@ To see changes in Javadoc, run:
 
 ```
-mkdocs gh-deploy
+dev/deploy-site.sh
````
should this be `./dev/deploy-site.sh`?

cwsteinbach replied:
@jackye1995  `dev/deploy-site.sh` and `./dev/deploy-site.sh` seem equivalent. Did you mean `../dev/deploy-site.sh`?

kbendick replied:
I do believe they're equivalent, but I also usually see commands referenced with an initial `./` when there's a directory in it.

jackye1995 replied:
Given the script works anywhere, I think we can instead just say "you can publish the site by executing the script `dev/deploy-site.sh`"

kbendick commented on dev/deploy-site.sh:
```diff
@@ -0,0 +1,62 @@
+rm -rf site
+mkdir site
+cp ../.asf.yaml site/
+mkdocs gh-deploy --dirty -r ${remote}
```
What does the `--dirty` flag do? Do we need it every time? Might be worth putting a comment on why we need it (for example, in case we don't need it in the future).

cwsteinbach replied:
@rdblue provided an explanation here: https://github.com/apache/iceberg/pull/2998#discussion_r693517612
I will update the script to include an explanation.

jackye1995 commented on dev/deploy-site.sh:
```diff
@@ -0,0 +1,62 @@
+trap "popd" EXIT
+
+# If the following commands stop working, please check this ASF INFRA
+# page for updates: https://infra.apache.org/asfyaml-mkdocs.html
```
which line would you expect to not work? If it's only the last line, I think we can consider echoing this error message if the command fails.

rdblue commented:
Can you also remove the site deploy tasks from gradle since this replaces them?

cwsteinbach commented:
@kbendick @jackye1995 @rdblue I decided that fixing the `deploySite` Gradle task makes more sense than replacing it with a new shell script. Consequently, in the latest version of this patch I have removed `dev/deploySite.sh` and added changes to the `deploySite` task. Please let me know if you approve of this approach. If so, I will then update the commit message and title of this PR to reflect these changes.

rdblue commented:
Can you update the PR title and description? Looks like those are out of date.

jackye1995 commented on tasks.gradle:
```diff
@@ -54,6 +54,18 @@ task refreshJavadoc(type: Exec) {
+  // is necessary.
+  commandLine('rm', '-rf', 'site')
+  commandLine('cp', '../.asf.yaml', 'site/')
+  commandLine('mkdocs', 'gh-deploy', '--dirty', '-r', remoteName)
```
can you match the original syntax without brackets? like `commandLine 'rm', '-rf', 'site'`
